src/utils/formModeUtils.ts

```diff
@@ -0,0 +1,158 @@
+export type FormMode = "Unauthorized" | "Edit" | "View Only" | "Review";
+export type FormStatus = "New" | "Submitted" | "In Review" | "Approved" | "In Progress" | "Rejected";
+
+export const EditStatuses = ["New", "In Progress", "Rejected"];
+export const ReviewStatuses = ["Submitted", "In Review"];
+export const FormModes = {
+  UNAUTHORIZED: "Unauthorized",
+  EDIT: "Edit",
+  VIEW_ONLY: "View Only",
+  REVIEW: "Review",
+} as const;
+
+/**
+ * Calculate the form mode for a user
+ * NOTE:
+ *  - This is a private helper function for getFormMode
+ *
+ * @param {FormStatus} formStatus - Current form status.
+ * @param {boolean} formBelongsToUser - True if the form belongs to the user.
+ * @returns {FormMode} - Form mode corresponding to the given form status and user.
+ */
+const _getFormModeForUser = (
+  user: User,
+  data: Application
+): FormMode => {
+  const userOrgRoleExists = user?.organization?.orgRole?.length > 0;
+  const belongsToSameOrg = user?.organization?.orgID === data?.organization?._id;
+  if ((user?.role !== "User" && !userOrgRoleExists) || (userOrgRoleExists && belongsToSameOrg && user.organization?.orgRole !== "Submitter")) {
+    return FormModes.UNAUTHORIZED;
+  }
+
+  const { status: formStatus } = data || {};
+  const formBelongsToUser = data?.applicant?.applicantID === user?.["_id"];
+  const isStatusViewOnlyForUser = ["Submitted", "In Review", "Approved"].includes(formStatus);
+
+  if (formStatus !== "New" && !formBelongsToUser) {
+    return FormModes.UNAUTHORIZED;
+  }
+  if (isStatusViewOnlyForUser) {
+    return FormModes.VIEW_ONLY;
+  }
+  if (EditStatuses.includes(formStatus)) {
+    return FormModes.EDIT;
+  }
+  return FormModes.UNAUTHORIZED;
+};
+
+/**
+ * Calculate the form mode for a Federal Lead
+ * NOTE:
+ *  - This is a private helper function for getFormMode
+ *
+ * @param {User} user - The current user
+ * @param {Application} data - The current application/submission
+ * @returns {FormMode} - Form mode corresponding to the given form status for a Federal Lead.
+ */
+const _getFormModeForFederalLead = (
+  user: User,
+  data: Application
+): FormMode => {
+  if (user?.role !== "FederalLead") {
+    return FormModes.UNAUTHORIZED;
+  }
+
+  const { status: formStatus } = data || {};
+  const isStatusViewOnlyForLead = ["In Progress", "Approved", "Rejected"].includes(formStatus);
```

Comment:
Federal Leads should basically never be unauthorized to view at a minimum. For now, the `New` status should be added here.

Reply:
Removed the check entirely and set default fallback to view only.